src/dataflow/src/sink/kafka.rs

```diff
@@ -1099,6 +1079,8 @@ where
     // our internal state after the send loop
     let mut progress_update = None;
 
+    let shutdown_flush = AtomicBool::new(false);
```

Comment:
No need for the atomic here, you can just use a `Cell<bool>`, but also `Relaxed` ordering is stressing me out (not relaxing at all!)